packages/controller/src/tasks/syncTenants.ts

```diff
@@ -21,7 +21,58 @@ import { KubeConfig } from "@kubernetes/client-node";
 import { set as updateTenants, Tenants } from "@opstrace/tenants";
 import dbClient, { ClientResponse } from "../dbClient";
 import { State } from "../reducer";
-import { log, SECOND } from "@opstrace/utils";
+import { log, SECOND, httpcl } from "@opstrace/utils";
+import { Response as GotResponse } from "got";
+
+function* setDefaultAlertmanagerConfigIfEmpty(tenant: string) {
+  const res: GotResponse<string> = yield httpcl(
+    `http://alertmanager.cortex.svc.cluster.local/api/v1/alerts`,
+    {
+      method: "GET",
+      headers: {
+        "X-Scope-OrgID": tenant
+      },
+      throwHttpErrors: false
+    }
+  );
+
+  if (res.statusCode > 499) {
+    // Network error, don't potentially overwrite existing config
```

Comment:
When we land here we did send the request and got a response. So, "network error" is an unexpected explanation.

Reply:
I set the `httpclient` to not throw on error, so we can control the flow easier for the case above where a 404 will be returned if alertmanager config the tenant hasn't been set yet.

Reply:
Hm, I am not sure if I got my point across.

Again, when we are in this code block it means that we got a response, i.e. no network error happened. Yet, in the code block it says ` // Network error, ...` and I cannot understand what is meant by that.

> I set the httpclient to not throw on error

I had seen that. It just means that an HTTP error response will not result in an error thrown. However, all errors resulting in the request not being sent or in no response being received (transient network errors, leading to DNS resolution errors or TCP connect()/recv() timeout etc) will nevertheless result in an error thrown.

My confusion is resolved when we remove the comment `// Network error, don't potentially overwrite existing config`.

Maybe a better explanation would be

```
// Handle 5xx responses: ambiguous result, we do not know if
// a config has already been set for this tenant or not. Back out.
```

Reply:
PR with suggested changes: https://github.com/opstrace/opstrace/pull/902